src/test/ui/issues/async-generator.rs

```diff
@@ -0,0 +1,7 @@
+#![feature(generators)]
+// edition:2018
+// Regression test for #67158.
```

Comment:
Let's move this test into `ui/generator/` and add the issue name to the file name?